cmake/OpenCVCompilerOptimizations.cmake

```diff
@@ -337,10 +341,10 @@ elseif(ARM OR AARCH64)
     endif()
     ocv_update(CPU_FP16_IMPLIES "NEON")
   else()
-    ocv_update(CPU_KNOWN_OPTIMIZATIONS "NEON;FP16")
+    ocv_update(CPU_KNOWN_OPTIMIZATIONS "NEON;FP16;DOTPROD")
```

Comment:
Same comment for here.
If you added `DOTPROD` in the list, you need to add corresponding compiler flag below.
This was tricky, since there wasn't such a "flag" in OpenCV for Armv8 till now.
You're the first one to add it.